inferno-types/src/Inferno/Types/Syntax.hs

```diff
@@ -89,6 +89,30 @@ module Inferno.Types.Syntax
     hideInternalIdents,
     substInternalIdents,
     getIdentifierPositions,
+    ImplType (..),
+    Namespace (..),
+    TCScheme (..),
+    TypeClass (..),
+    TypeClassShape (..),
+    TypeMetadata (..),
+    Substitutable (..),
+    Subst (..),
+    Scheme (..),
+    (.->),
+    sch,
+    var,
+    tySig,
+    namespaceToIdent,
+    typeInt,
+    typeBool,
+    typeDouble,
+    typeWord16,
+    typeWord32,
+    typeWord64,
+    typeText,
+    typeResolution,
+    typeTimeDiff,
+    typeTime,
```

Comment:
All these type declarations had to move into this file from `Type.hs` because `Expr` now depends on `TCScheme` (because the new `LetAnnot` constructor holds a type annotation). Is that okay?